### PR TITLE
barista(overview-pages): Order tiles on overview pages by title.

### PR DESCRIPTION
**Pull request by NinaKammerlander**
### <strong>Pull Request</strong>

<hr>
Hi, thank you for contributing to Barista with this pull request (PR).

To ensure a fast process and merging of your PR please make sure it fulfills the
coding standards and contribution guidelines.

- A feature proposal has been provided, discussed and approved first.
- There is a meaningful description of the issue in GitHub (Screenshots are
  often helpful).
- If the PR introduces breaking-changes or deprecations it matches the following
  guidelines.
  - The commit message follows our commit guidelines.
  - Tests for the changes have been added (for bug fixes / features).
  - Docs have been added / updated (for bug fixes / features).

Please choose the type appropriate for the changes below: <br>

#### Checklist

- [ ] I have read the CONTRIBUTING doc and I follow the PR guidelines
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: tools/barista/src/builder/overview.ts [hunk 1]
```diff
@@ -41,6 +41,41 @@ const highlightedItems = [
   'Theming',
 ];
 
+function orderSectionItems(categoryNav: BaOverviewPage): BaOverviewPage {
+  for (const section of categoryNav.sections) {
+    const sectionItems = section.items;
+    if (section.title === 'Documentation') {
```
Comment (lara-aigmueller) [inline tools/barista/src/builder/overview.ts]: Also other pages (not only documentation) can have the order property set. Sort all overview items the same way: check if both items have an order-value and compare this one, if not, sort by title, if only one has an order-value, this one should come first.

File: tools/barista/src/generators/category-navigation.ts [hunk 2]
```diff
@@ -108,6 +108,40 @@ function addSidenavToPages(
...
+
+      if (b.order) {
+        return 1;
+      }
+
+      if (a.order) {
+        return -1;
+      }
+
+      if (a.title < b.title) {
```
Comment (lara-aigmueller) [inline tools/barista/src/generators/category-navigation.ts]: Please use `a.localeCompare(b)` instead: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare

Comment (tomheller) [reply tools/barista/src/generators/category-navigation.ts]:
```suggestion
      if (a.title && b.title) {
        return a.title.localeCompare(b.title)
      }
```

File: tools/barista/src/generators/category-navigation.ts [hunk 3]
```diff
@@ -242,42 +261,23 @@ export const overviewBuilder = async () => {
...
-        JSON.stringify(componentOverview, null, 2),
-        {
-          flag: 'w', // "w" -> Create file if it does not exist
-          encoding: 'utf8',
-        },
-      );
-    }
+    const overviewfilepath = join(DIST_DIR, `${directory}.json`);
+    // Write file with page content to disc.
+    // tslint:disable-next-line: no-magic-numbers
```
Comment (tomheller) [inline tools/barista/src/generators/category-navigation.ts]: please remove tslint disable, there is no magic number there any more.

**Pull request merged**